x-pack/plugins/security_solution/server/endpoint/services/actions/action_list.ts

```diff
@@ -157,6 +168,17 @@ const getActionDetailsList = async ({
     results: actionResponses?.body?.hits?.hits,
   });
 
+  let hosts: ActionDetails['hosts'];
+  let agentsHostInfo: { [id: string]: string };
+  if (showHostsInfo) {
+    // get host metadata info with queried agents
+    agentsHostInfo = await getAgentHostNamesWithIds({
```

Comment:
(Optional) This API could be a bit more efficient by making it concurrently with the API above for the action responses (`using `Promise.all()`)